scripts/simulate.py

```diff
@@ -0,0 +1,192 @@
+"""
+CLI for hera_sim
+"""
+import argparse
+import copy
+import os
+import sys
+import warnings
+import yaml
+
+import hera_sim
+from hera_sim import cli_utils
+from pyuvsim.simsetup import _parse_layout_csv
+from astropy.coordinates import Angle
+from astropy import units
+try:
+    import bda
+    from bda import bda_tools
+except ImportError:
+    warnings.warn("bda failed to import. Baseline-dependent averaging not available.")
+    bda = None
+
+parser = argparse.ArgumentParser(
+    description="Run a hera_sim-managed simulation from the command line."
+)
+parser.add_argument("config", type=str, help="Path to configuration file.")
+parser.add_argument(
+    "-o",
+    "--outfile",
+    type=str,
+    default=None,
+    help="Where to save simulated data. Overrides outfile specified in config."
+)
+parser.add_argument(
+    "-v",
+    "--verbose",
+    default=False,
+    action="store_true",
+    help="Print progress updates."
+)
+parser.add_argument(
+    "-sa",
+    "--save_all",
+    default=False,
+    action="store_true",
+    help="Save each simulation component."
+)
+parser.add_argument(
+    "--clobber",
+    default=None,  # To allow the user to specify clobber in the config file.
+    action="store_true",
+    help="Overwrite existing files in case of name conflicts."
+)
+args = parser.parse_args()
+
+if args.verbose:
+    print("Reading configuration file and validating contents...")
+
+with open(args.config, 'r') as cfg:
+    config = yaml.load(cfg.read(), Loader=yaml.FullLoader)
+
+cli_utils.validate_config(config)
+bda_params = config.get("bda", {})
+if bda_params:
+    bda_params["corr_FoV_angle"] = Angle(
+        bda_params.get("corr_FoV_angle", 20 * units.deg)
+    )
+    
+if bda_params and bda is None:
+    raise ModuleNotFoundError("Please ensure bda is installed and try again.")
+
+filing_params = cli_utils.get_filing_params(config)
+args.outfile = args.outfile or os.path.join(
+    filing_params["outdir"],
+    filing_params["outfile_name"]
+)
+args.clobber = filing_params["clobber"] if args.clobber is None else args.clobber
+if os.path.exists(args.outfile) and not args.clobber:
+    print(f"File exists and clobber=False; exiting.")
+    sys.exit()
+
+if args.verbose:
+    print("Preparing simulation...")
+
+instrument_parameters = {}
+defaults = config.get("defaults", None)
+if defaults is not None:
+    # TODO: update the defaults module to handle validating the choice of defaults
+    if not isinstance(defaults, str):
+        msg = "Defaults must be specified using a string. The string may be either "
```

Comment:
You can literally do `ValueError("Defaults must be...")` and put new lines between each line -- as long as it's in brackets, strings by default concatenate between lines.

Reply:
Yep... I think this was more of a "this looks nicer to me" thing? I'm not sure though. I seem to recall having an issue with something involving breaking up long strings across lines, but that may have been with `warnings.warn`, rather than any built-in exceptions.

Reply:
This value-checking block has been removed, since it is handled in `cli_utils.validate_config`.